### ISSUE TITLE
suggestions for apps that use Web Workers ?

### ISSUE THREAD

**Issue by jokeyrhyme** [davezuko/react-redux-starter-kit#879]
I'm starting from a 3.0 alpha commit: https://github.com/davezuko/react-redux-starter-kit/commit/fdb579de48b44efa2306f3c2f2dca71691db741a

I'd like to set this up to output an additional JavaScript file called "worker.js".

What I've done so far:

- new "src/worker.js" file

- updated "build/webpack.conf.js"  to include it in `webpackConfig.entry`

    ```js
    webpackConfig.entry = {
      app: __DEV__
        ? APP_ENTRY_PATHS.concat(`webpack-hot-middleware/client?path=${config.compiler_public_path}__webpack_hmr`)
        : APP_ENTRY_PATHS,
      vendor: config.compiler_vendor,
      worker: paths.client('worker.js')
    }
    ```

This does compile and produce a new "dist/worker.HASH.js" file as expected.

However, in both NODE_ENV=production and NODE_ENV=development, the resulting "index.html" file now also includes "worker.js" via a script tag. This is not my intention, as this file needs to be loaded via a JavaScript API later. I want this code to execute in a Web Worker, not in the main context / frame / process / thread / thing.

I've looked through the webpack configuration and build scripts, and I no naive tweaks seem to stop the new script tag from appearing in "index.html". Any ideas?

**Comment by jokeyrhyme** [comment]
This same approach will likely also benefit apps that use Service Workers, too.

**Comment by twelch** [comment]
@jokeyrhyme  Might be of help.  I use a library called mapbox-gl that uses web workers (I didn't develop).  They use a package called webworkify-webpack.  Can take a look at their config and worker setup.

https://github.com/mapbox/mapbox-gl-js/blob/master/webpack.config.example.js
More context - https://github.com/mapbox/mapbox-gl-js/issues/1649

**Comment by jokeyrhyme** [comment]
Yeah, I think the way to do this is to have a second webpack config. I can probably just have a file that imports / clones the useful parts of the primary webpack config.
